From 264c7981e8ea8ff7fdd502a58a3e5195b8594630 Mon Sep 17 00:00:00 2001
From: WangJing <wangjing@mprc.pku.edu.cn>
Date: Mon, 10 Oct 2011 17:14:04 +0800
Subject: [PATCH 015/641] UniCore64:Add definition of ELF_ET_DYN_BASE

Fixed error:
fs/binfmt_elf.c:800:16: error: 'ELF_ET_DYN_BASE' undeclared (first use in this function)

Signed-off-by: WangJing <wangjing@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/elf.h |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/unicore64/include/asm/elf.h b/arch/unicore64/include/asm/elf.h
index ccc44ca..ff083cb 100644
--- a/arch/unicore64/include/asm/elf.h
+++ b/arch/unicore64/include/asm/elf.h
@@ -34,4 +34,11 @@ typedef struct fpu_regs elf_fpregset_t;
 
 #define SET_PERSONALITY(ex) set_personality(PER_LINUX)
 
+/* This is the location that an ET_DYN program is loaded if exec'ed.  Typical
+   use of this is to invoke "./ld.so someprog" to test out a new version of
+   the loader.  We need to make sure that it is out of the way of the program
+   that it will "exec", and that there is sufficient room for the brk.  */
+
+#define ELF_ET_DYN_BASE	(2 * TASK_SIZE / 3)
+
 #endif /* __ASM_UNICORE64_ELF_H__ */
-- 
1.7.9.5

